blockprotocol/blockprotocol
# Add "Change payment method" modal to upgrade subscription page

## 🌟 What is the purpose of this PR?




This PR adds a "Change payment method" modal to the upgrade subscription page.

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1203623179643281/1203880493107349/f) _(internal)_

## 🔍 What does this change?




See commits.

## 📜 Does this require a change to the docs?



Not that I'm aware of.

## 🛡 What tests cover this?



Manual testing.

## ❓ How to test this?



1. Checkout the branch
2. Add the `NEXT_PUBLIC_BILLING_FEATURE_FLAG=1` env var to your `app/site/.env.local` to enable the billing feature flag
3. Run the local developer environment for the BP site
4. Checkout the HASH internal API repo
5. Follow the instructions in the internal API README to run the local developer environment, except for the mongo db external service (as the BP site is already running mongo db)
6. Login as a user and navigate to `/settings/billing`
7. Ensure the logged in user is subscribed to the "hobby" subscription tier
8. Go to the "upgrade subscription" page at `/settings/billing/upgrade?tier=pro`, and use the "change" button next to the default payment method to open the "Change payment method" modal

## 📹 Demo



https://user-images.githubusercontent.com/42802102/217347042-b85b24a2-7961-4c1b-97e4-80dd81bee5f9.mov
